Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -677,7 +681,7 @@ public class PBXProjGenerator {
 
         let targetSupportsDirectEmbed = !(target.platform.requiresSimulatorStripping &&
             (target.type.isApp || target.type == .watch2Extension))
-        let directlyEmbedCarthage = target.directlyEmbedCarthageDependencies ?? targetSupportsDirectEmbed
+        let directlyEmbedCarthage = target.directlyEmbedCarthageDependencies ?? project.options.carthageXCFrameworks || targetSupportsDirectEmbed
```

Comment:
↓ may be corect?

```suggestion
        let directlyEmbedCarthage: Bool
            
        if let directlyEmbedCarthageDependencies = target.directlyEmbedCarthageDependencies {
            directlyEmbedCarthage = directlyEmbedCarthageDependencies
        } else if project.options.carthageXCFrameworks {
            directlyEmbedCarthage = project.options.carthageXCFrameworks
        } else {
            directlyEmbedCarthage = targetSupportsDirectEmbed
        }
```

This makes pass one failed test